adoption-metrics/define_adoption_metrics.md

```diff
@@ -0,0 +1,56 @@
+# What adoption metrics Packit team wants to measure
+
+## Static meaurement (updated periodically)
+
+### Newcomers adoption
+
+- Total amount of newcomers in a given period (monthly/quarterly/half yearly/yearly).
+- How many newcomers (percentage) from a given period already tried some functionality that
+  Packit(-service) provides (COPR, TF, all downstream jobs).
+- How fast newcomers are? How much time (in average) they need from approval to actually
+  do something?
+- How often are different services (COPR, TF, all downstream jobs) used by newcomers?
+  What they use most often and what they don't use at all?
+
+### Active users
+
+Active users are those from whom we have recorded at least one activity in the past period
+and they do not fit into the category of newcomers.
+
+- How often are different services (COPR, TF, all downstream jobs) used by active users?
+  What they use most often and what they use less?
+- How frequently active users use packit services?
+
+### Users retention
+
+- Percentage of previous newcomers who became active Packit users.
+  ((amount of newcomers from previous period becoming an active users
+  / amount of newcomers from previous period) \* 100)
+- What percentage of users are active users? (How many inactive users do we have?)
+
+### Usage of Packit services
+
+- Percentage representation of individual services e.g. (COPR requests / all requests) \* 100
+
+## The following metrics will be displayed as a histograms
+
+As the metrics above - these will be updated once a given period.
+
+- Show most 15/20 active project of ... COPR, TF, all downstream jobs
+
+It may look in very similar way how Packit team does its yearly evaluation. As an example
+I took a graph of [Top 20 projects in the number of PR Copr Builds in 2021](https://packit.dev/posts/2021-in-numbers/#top-20-projects-in-the-number-of-pr-copr-builds)
+
+![alt text](https://packit.dev/images/2021-in-numbers-top20-copr-builds.png)
+
+## Real time metrics
+
+This is heavily inspired by [COPR's real time stats](https://copr.fedorainfracloud.org/status/stats/)
+
+Real time graph will show overall activity of users in a given period (day/week/quarter)
+
+- Make a graph for each service packit provides (COPR, TF, all downstream jobs) and display
+  activity. This may help to see and identify peaks and lows in overall activity of users.
```

Comment:
yes please!

Reply:
The real-time data in Copr are for admins to identify issues. It may be hard to set up warning levels. E.g. there are no problems if there appear 4k actions, but because actions are mostly no-op, it should go down almost immediately. If not, it may indicate a problem. Or - there is no problem if there are 40k builds in a queue, but the chart should decline. If not, it may indicate a problem in a scheduler or spawning builds.  So for real-time data, it is useful to plot data, which helps admin spot anomalies in the behaving of the whole system.